torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp

```diff
@@ -117,12 +117,13 @@ const std::vector<Expr*>& Val::uses() const {
 // for index expressions.
 void Val::resolveIndexDtype() {
   TORCH_INTERNAL_ASSERT(
-      vtype_ == ValType::TensorView || vtype_ == ValType::Scalar,
+      vtype_ == ValType::TensorView || vtype_ == ValType::Scalar ||
+          vtype_ == ValType::NamedScalar,
```

Comment:
Misaligned vectorize is always good at surprising me 😕

Reply:
Why does `MisalignedVectorize` have anything to do with?

Reply:
Because `MisalignedVectorize` does things like `named_scalar = expression`, but a lot of our system is assuming that named scalar is just itself, and it has no definition. That's not a big issue, just easy mechanical fixes. Just it is a bit annoying...